File: aurora-core/src/main/java/net/gree/aurora/domain/cluster/TaggedClusterIdResolverImpl.java
package net.gree.aurora.domain.cluster;

import org.sisioh.dddbase.model.impl.AbstractEntity;

final class TaggedClusterIdResolverImpl<HINT>  extends AbstractEntity<TaggedClusterIdResolverId> implements TaggedClusterIdResolver<HINT> {

    private final ClusterIdResolver<HINT> clusterIdResolver;

    /**
     * インスタンスを生成する。
     *
     * @param identity {@link org.sisioh.dddbase.model.Identity}
     * @param clusterIdResolver
     */
    public TaggedClusterIdResolverImpl(TaggedClusterIdResolverId identity, ClusterIdResolver<HINT> clusterIdResolver) {
        super(identity);
        this.clusterIdResolver = clusterIdResolver;
    }

    @Override
    public ClusterIdResolver<HINT> getClusterIdResolver() {
        return clusterIdResolver;
    }

}

File: aurora-scala/src/main/scala/net/gree/aurora/scala/domain/cluster/TaggedClusterIdResolverImpl.scala
package net.gree.aurora.scala.domain.cluster


import net.gree.aurora.domain.cluster.{TaggedClusterIdResolver => JTaggedClusterIdResolver}

class TaggedClusterIdResolverImpl[HINT](val underlying: JTaggedClusterIdResolver[HINT])
  extends TaggedClusterIdResolver[HINT] {
  val identity: TaggedClusterIdResolverId = TaggedClusterIdResolverId(underlying.getIdentity.getValue)
  val clusterIdResolver: ClusterIdResolver[HINT] = ClusterIdResolver(underlying.getClusterIdResolver)
}
